src/test/java/dk/kb/labsapi/ImageExportTest.java

```diff
@@ -247,5 +261,57 @@ public void testAvoidZipClose() throws IOException {
         zos.closeEntry();
     }
 
+    @Test
+    public void testStreaming() throws IOException {
+        String query = "hest";
+        int startTime = 1750;
+        int endTime = 1780;
+        int max = 10;
+        SolrQuery finalQuery = ImageExport.getInstance().fullpageSolrQuery(query, startTime, endTime, max);
+        Stream<SolrDocument> docs = ImageExport.getInstance().streamSolr(finalQuery, max);
+        // Get fullPage metadata
+        Stream<FullPageMetadata> pageMetadata = docs.map(ImageExport.getInstance()::streamMetadataForFullPage);
+
+        long processed = pageMetadata.count();
+        assertEquals(10, processed);
+    }
+
+    @Test
+    public void testStreamingIllustrations() throws IOException {
+        ImageExport exporter = ImageExport.getInstance();
+        String query = "hest";
+        int startTime = 1750;
+        int endTime = 1780;
+        int max = 10;
+        SolrQuery finalQuery = exporter.illustrationSolrQuery(query, startTime, endTime, max);
+        Stream<SolrDocument> docs = exporter.streamSolr(finalQuery, max);
+        HashSet<String> uniqueUUIDs = new HashSet<>();
+        Stream<IllustrationMetadata> illustrationMetadata = docs.flatMap(doc -> exporter.streamMetadataForIllustrations(doc, uniqueUUIDs));
+
+        illustrationMetadata.forEach(Assertions::assertNotNull);
```

Comment:
You can combine checks of individual elements in the stream with overall check for the number of elements received:

```
        long received = illustrationMetadata.
                peek(Assertions::assertNotNull).
                count();
        assertTrue(received > 0, "Some results should be received");
```